src/components/ProgressAnimation/index.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Progress animation component.
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { purple, gray } from '@carbon/colors';
+
+import { getComponentNamespace } from '../../globals/namespace';
+
+const namespace = getComponentNamespace('progress-animation');
+
+function ProgressAnimation({
+  children,
+  className,
+  id,
+  filledColor,
+  unfilledColor,
+  percentage,
+  animationTimer,
+}) {
+  const progressAnimationName = `progress-animation-${id}`;
+
+  return (
+    <div className={classnames(namespace, className)}>
+      {/**
+       * If `animationTimer` is `null`, no need to insert a style tag.
+       * Otherwise, a new style tag with a unique keyframe needs to be
+       * created per component instance, because the unique progress
+       * percetange is used in the keyframe animation.
+       */}
+      {animationTimer ? (
+        <style
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{
+            __html: `
+              @keyframes ${progressAnimationName} {
```

Comment:
For an animated instance of this component, it is required that it is animated to the percentage value provided. Which to me, means that the svg path needs to move from a "0" value to the provided percentage complete value.

To achieve this with keyframes (which seemed like the best solution? let me know if you disagree), we need to insert a style tag in this manner so that *each* instance of the component has each own keyframes, based on its own unique percentage complete value.

Worth noting that I did explore using `emotion` and `styled-components`, but I don't believe those solution are ideal for us. We need these keyframes to be in the global scope AND also have the prop `percentage` value. Hence, the solution above. 🤔 But please let me know what you think!